### PR TITLE
Add tilt dev guide

### PR DESCRIPTION
**Pull request by Joll59**
**What this PR does / why we need it**:

adding minor changes to incorporate tilt dev workflow. 

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes # 150

**Special notes for your reviewer**:

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]

File: docs/DEVELOP.md [hunk 1]
```diff
@@ -1 +1,12 @@
...
+
+- working within a `codespace` you can install and instantiate a kind cluster for testing
+
+  - Follow instructions on `kind` website for install from binary.
+  - Move kind binary to `/usr/bin`:`mv ./kind /usr/bin`
+  - kind should now be installed and ready
+  - `kind create cluster` should have you up and running _Kind switches `kubectl` context to `kind-kind` so you can work with your LOCAL kind cluster using kubectl_
+
+- installing tilt is optional.
+  - `curl -fsSL https://raw.githubusercontent.com/tilt-dev/tilt/master/scripts/install.sh | bash`
```
Comment (sozercan) [inline docs/DEVELOP.md]: let's link to tilt docs here

File: docs/DEVELOP.md [hunk 2]
```diff
@@ -1 +1,12 @@
-Developing this project requires access to a Kubernetes cluster and Go version 1.16 or later.
\ No newline at end of file
+Developing this project requires access to a Kubernetes cluster and Go version 1.16 or later.
+
+- working within a `codespace` you can install and instantiate a kind cluster for testing
+
+  - Follow instructions on `kind` website for install from binary.
+  - Move kind binary to `/usr/bin`:`mv ./kind /usr/bin`
```
Comment (sozercan) [inline docs/DEVELOP.md]: can we link to kind docs here for installation? folks might use a package manager so they don't necessarily need to move the binary

File: Tiltfile.sample [hunk 3]
```diff
@@ -0,0 +1,127 @@
+# Welcome to Tilt!
```
Comment (sozercan) [inline Tiltfile.sample]: is sample used anywhere? should we keep it?

File: Tiltfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+""".strip())
+
+docker_build('eraser-build', context='.')
+
+k8s_yaml('./deploy/eraser.yaml')
+# add further deployments above. Reference sample tilt file. or https://docs.tilt.dev/install.html
+
+local_resource('Kubectl get pods',cmd='kubectl get pods -A')
+local_resource('Kubectl get imageList',cmd='kubectl get ImageList -A')
+local_resource('Kubectl describe imageList',cmd='kubectl describe ImageList imagelist -A')
```
Comment (sozercan) [inline Tiltfile]: not familiar with these, what are the local resource for?

File: Tiltfile.sample (hunk 3 above)
Comment (Joll59) [reply Tiltfile.sample]: I was keeping the sample as reference. I can see the case for and against it. I'll remove it for now.

File: Tiltfile (hunk 4 above)
Comment (Joll59) [reply Tiltfile]: They are simply `kubectl` commands. These are the commands in the Getting started, I included as samples of commands that can be utilized while using a tilt file.

File: Tiltfile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+## https://docs.tilt.dev/install.html
+### This is a python file. Change Text editor file association for Syntax highlighting.
+
+print("""
+-----------------------------------------------------------------
+ERASER(📒 ✏️) Assumes docker installed, a kind cluster instantiated and kubectl in the right context. More on Tilt: https://docs.tilt.dev/install.html
+-----------------------------------------------------------------
+""".strip())
+
+docker_build('eraser-build', context='.')
```
Comment (sozercan) [inline Tiltfile]: I am a little confused, what is `eraser-build` here? how do we build a new image and push to kind registry here?

example: https://github.com/open-policy-agent/gatekeeper/pull/1895/

File: Tiltfile [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+k8s_yaml('./deploy/eraser.yaml')
+
+# add further deployments above.
+
+local_resource('Kubectl get pods',cmd='kubectl get pods -A')
+local_resource('Kubectl get imageList',cmd='kubectl get ImageList -A')
```
Comment (sozercan) [inline Tiltfile]: imagelist is a cluster scoped CRD, no need for -A

File: Tiltfile (hunk 5 above)
Comment (sozercan) [reply Tiltfile]: wouldn't we just deploy 0.1.0 every time since pull policy is always?

**Review by sozercan — CHANGES_REQUESTED** [review]
see comments

File: Tiltfile (hunk 5 above)
Comment (Joll59) [reply Tiltfile]: You are correct, I addressed in latest update.

File: kindLocalRegistry.sh [hunk 7]
```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
```
Comment (sozercan) [inline kindLocalRegistry.sh:1]: looks like this file needs to be an executable

File: Tiltfile [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Eraser Tiltfile.
+
+print("""
+-----------------------------------------------------------------
+ERASER(📒 ✏️) Assumes docker installed, a kind cluster instantiated and kubectl in the right context.
+-----------------------------------------------------------------
+""".strip())
+
+# build image and push into local registry.
+docker_build('localhost:5001/eraser-local', context='.')
```
Comment (sozercan) [inline Tiltfile:10]: this is building manager only, we need to build eraser too

File: Tiltfile [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+objects = read_yaml_stream('./deploy/eraser.yaml')
+
+## point deployment yaml at local image in registry.
+for o in objects:
+    if o['kind'] == 'Deployment':
+        o['spec']['template']['spec']['containers'][0]['args'][1]= "--eraser-image=localhost:5001/eraser-local"
```
Comment (sozercan) [inline Tiltfile:17]: these won't be the same image, we'll need to build eraser seperately

**Review by sozercan — CHANGES_REQUESTED** [review]

**Comment by sozercan** [comment]
Looks like this PR went stale. Closing for now, feel free to reopen if still interested in picking this up.

**Pull request closed without merging**